### ISSUE TITLE
Math function `ACOS` does not throw error outside [-1,1]

### ISSUE THREAD

**Issue by manticore-projects** [duckdb/duckdb#11442]
### What happens?

`ACOS(x)` (and `ASIN(x)`) should throw `ERROR` for `x<-1` or `x>1`. 
But `ACOS(-2)` and `ACOS(2)` return `NAN` while only  `ASIN(-2)` or `ASIN(2)` correctly throw errors.

### To Reproduce

```sql

SELECT acos(-2) a, acos(-1) b, acos(0) b, acos(1) d, acos(2) d;

/* returns
┌─────┬─────────┬────────┬──────┬─────┐
│ a   │ b       │ b      │ d    │ d   │
├─────┼─────────┼────────┼──────┼─────┤
│ NaN │ 3.14159 │ 1.5708 │ 0.00 │ NaN │
└─────┴─────────┴────────┴──────┴─────┘
*/


### OS:

Linux

### DuckDB Version:

0.10.1

### DuckDB Client:

Java

### Full Name:

Andreas Reichel

### Affiliation:

manticore-projects.com

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by soerenwolfers** [comment]
I think the opposite: duckdb should never throw on floating-point operations and always return NaN / +-infinity instead.
See https://github.com/duckdb/duckdb/discussions/10956

I agree it shouldn't be inconsistent though. 

FWIW, it's not inconsistent for m (Linux, duckdb 0.10.2.dev311): It throws for all of for `acos(2)` / `asin(2)` / `sqrt(-2)` / `sin('Infinity')` (confusingly it throws `InvalidInput` for the former two and `OutOfRange` for the latter two.)

**Comment by manticore-projects** [comment]
Thank you for the quick round-turn, I will switch to latest Snapshot and close this.
I am also not worried about Error vs. NAN -- just about consistency. I will keep hitting such stuff since I am testing all standard RDBMS functions against DuckDB. 

Btw, any particular reason why the Hyperbolicus functions are missing? Would you accept PR's on those or was it not ins cope for a reason? (they are commonly supported on most RDBMS, likely because of GEO I assume.)

**Comment by manticore-projects** [comment]
Confirmed to be fixed in 0.11-Snapshot yet.
